src/components/table/EnhancedTable.jsx

```diff
@@ -130,96 +118,112 @@ const EnhancedTable = ({
   }, [fetchData, pageIndex, pageSize]);
 
   return (
-    <TableContainer>
+    <>
       {shouldVisibleToolbar ? (
         <TableToolbar
-          modalOpen={modalOpen}
-          handleCloseModal={handleCloseModal}
-          handleClickOpen={handleClickOpen}
-          deleteUserHandler={deleteUserHandler}
-          addUserHandler={formAction}
-          tableTitle="Page Config"
-        >
-          {formDialog}
-        </TableToolbar>
+          preGlobalFilteredRows={preGlobalFilteredRows}
+          setGlobalFilter={setGlobalFilter}
+          globalFilter={globalFilter}
+        />
       ) : null}
-      <MaUTable {...getTableProps()}>
-        <TableHead>
-          {headerGroups.map((headerGroup, index) => (
-            <TableRow key={index} {...headerGroup.getHeaderGroupProps()}>
-              {headerGroup.headers.map((column, index) => (
-                <TableCell
-                  key={index}
-                  {...(column.id === "selection"
-                    ? column.getHeaderProps()
-                    : column.getHeaderProps(column.getSortByToggleProps()))}
-                  style={{ whiteSpace: "nowrap" }} // Ensure content does not wrap
-                >
-                  {column.render("Header")}
-                  {column.id !== "selection" ? (
-                    <TableSortLabel
-                      active={column.isSorted}
-                      // react-table has a unsorted state which is not treated here
-                      direction={column.isSortedDesc ? "desc" : "asc"}
-                    />
-                  ) : null}
-                </TableCell>
-              ))}
-            </TableRow>
-          ))}
-        </TableHead>
-        <TableBody>
-          {page.map((row, i) => {
-            prepareRow(row);
-            return (
-              <TableRow key={i} {...row.getRowProps()}>
-                {row.cells.map((cell, index) => {
-                  if (index === row.cells.length - 1) {
-                    // Render actions in the last cell
-                    return (
-                      <TableCell key={index}>
-                        <IconButton onClick={() => handleEdit(row)}>
-                          <EditIcon />
-                        </IconButton>
-                        <IconButton onClick={() => handleDelete(row)}>
-                          <DeleteIcon />
-                        </IconButton>
-                      </TableCell>
-                    );
-                  } else {
-                    // Render regular cell
-                    return (
-                      <TableCell key={index} {...cell.getCellProps()}>
-                        {cell.render("Cell")}
-                      </TableCell>
-                    );
-                  }
-                })}
+      <TableContainer>
+        <MaUTable {...getTableProps()}>
```

Comment:
Minor suggestion try to use Mui descriptive names
eg : instead of MaUTable we ca use MuiTable